pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -543,11 +547,33 @@ public TransactionInBufferStats getTransactionInBufferStats(TxnID txnID) {
     }
 
     @Override
-    public TransactionBufferStats getStats() {
+    public TransactionBufferStats getStats(boolean lowWaterMarks) {
         TransactionBufferStats transactionBufferStats = new TransactionBufferStats();
         transactionBufferStats.lastSnapshotTimestamps = this.lastSnapshotTimestamps;
         transactionBufferStats.state = this.getState().name();
         transactionBufferStats.maxReadPosition = this.maxReadPosition.toString();
+        if (lowWaterMarks) {
+            transactionBufferStats.lowWaterMarks = this.lowWaterMarks;
+        }
+        transactionBufferStats.ongoingTxns = ongoingTxns.size();
+
+        Optional<LookupResult> lookupResultOptional = Optional.empty();
+        try {
+            lookupResultOptional = this.topic
+                    .getBrokerService()
+                    .getPulsar()
+                    .getNamespaceService()
+                    .getBrokerServiceUrlAsync(TopicName.get(topic.getName()),
+                            LookupOptions.builder().loadTopicsInBundle(false).build())
+                    .get();
+        } catch (Exception e) {
+            // Just log the exception, nothing else to do
+            log.warn("getBrokerServiceUrl [{}]", e.getMessage(), e);
+        }
+        if (lookupResultOptional.isPresent()) {
+            LookupData lookupData = lookupResultOptional.get().getLookupData();
+            transactionBufferStats.brokerOwnerURL = lookupData.getBrokerUrl();
```

Comment:
I think that the client can do this lookup locally.
using Broker resources to do lookups may be expensive

this method is not supposed to be called frequently but if there is no strict need I would prefer to not do the lookup inside this method